api4/system.go

```diff
@@ -145,6 +147,39 @@ func testEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func testSiteURL(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	if *c.App.Config().ExperimentalSettings.RestrictSystemAdmin {
+		c.Err = model.NewAppError("testSiteURL", "api.restricted_system_admin", nil, "", http.StatusForbidden)
+		return
+	}
+
+	siteURLByte, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("testSiteURL", "app.admin.test_site_url.failure", nil, "", http.StatusBadRequest)
+		return
+	}
+	siteURL := string(siteURLByte)
+
+	if siteURL == "" {
+		c.Err = model.NewAppError("testSiteURL", "app.admin.test_site_url.failure", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	url := fmt.Sprintf("%s/api/v4/system/ping", siteURL)
+	res, err := http.Get(url)
+	if err != nil || res.StatusCode != 200 {
+		c.Err = model.NewAppError("testSiteURL", "app.admin.test_site_url.failure", nil, "", http.StatusBadRequest)
+		return
+	}
```

Comment:
Please move above [lines](https://github.com/mattermost/mattermost-server/pull/11944/files#diff-00da92b0e7d376d5c117d389f8021d87R173-R178) to `app` package since it's more of internal implementation.

Reply:
Good point, done